### PR TITLE
Zookeeper topology

### PR DESCRIPTION
**Pull request by endika-ravn**
Description of the change

This change enhances the chart to support Topology Spread Constraints for assignment pods spread across your cluster among failure-domains such as regions, zones, nodes, and other user-defined topology domains. This can help to achieve high availability as well as efficient resource utilization.

Benefits

Users have an optional to use Topology Spread Constraints

Possible drawbacks

None

Applicable issues

Additional information

Copied mostly from #6081

Checklist

Chart version bumped in Chart.yaml according to semver.
Variables are documented in the README.md

Title of the PR starts with chart name (e.g. [bitnami/chart])

⚠️ Keep in mind that if you want to make changes to the kubeapps chart, please implement them in the kubeapps repository. This is only a synchronized mirror.

### CODE REVIEW

File: bitnami/zookeeper/Chart.yaml [hunk 1]
```diff
@@ -21,4 +21,4 @@ name: zookeeper
 sources:
   - https://github.com/bitnami/bitnami-docker-zookeeper
   - https://zookeeper.apache.org/
-version: 7.1.2
+version: 7.2.0
```
Comment (marcosbc) [inline bitnami/zookeeper/Chart.yaml:24]: Could you fix the lint error? It is probably happening due to the text editor, if you simply save the file with another one (i.e. `vim` or `emacs`) it should fix the issue.

**Review by marcosbc — CHANGES_REQUESTED** [review]
Thanks for the contribution! Could you have a look at my comments? Apart from that it looks fine.

File: bitnami/zookeeper/values.yaml [hunk 2]
```diff
@@ -279,6 +279,10 @@ nodeSelector: {}
 ## Ref: https://kubernetes.io/docs/concepts/configuration/taint-and-toleration/
 ##
 tolerations: []
+## Topology Spread Constraints for pod assignment spread across your cluster among failure-domains. Evaluated as a template
```
[the remark this answers is not in the document]
Comment (endika-ravn) [reply bitnami/zookeeper/values.yaml]: added

File: bitnami/zookeeper/templates/statefulset.yaml [hunk 3]
```diff
@@ -74,6 +74,9 @@ spec:
       {{- if .Values.tolerations }}
       tolerations: {{- include "common.tplvalues.render" (dict "value" .Values.tolerations "context" $) | nindent 8 }}
       {{- end }}
+      { { - if .Values.topologySpreadConstraints } }
+      topologySpreadConstraints: { { - include "common.tplvalues.render" (dict "value" .Values.topologySpreadConstraints "context" .) | nindent 8 } }
+      { { - end } }
```
[the remark this answers is not in the document]
Comment (endika-ravn) [reply bitnami/zookeeper/templates/statefulset.yaml]: the editor auto style tricked me

**Review by marcosbc — APPROVED** [review]
LGTM, thanks for the contribution!

**Pull request merged**